web/source/osk/oskKey.ts

```diff
@@ -399,42 +399,19 @@ namespace com.keyman.osk {
         oldText;
     }
 
-
-    private unicodeKeyIdToString(id: string): string {
-      // This is similar to defaultOutput.ts:forUnicodeKeynames and could potentially
-      // be refactored in the future.
-      if(!id || id.substr(0,2) != 'U_') {
-        return null;
-      }
-
-      let result = '';
-      const codePoints = id.substr(2).split('_');
-      for(let codePoint of codePoints) {
-        const codePointValue = parseInt(codePoint, 16);
-        if (((0x0 <= codePointValue) && (codePointValue <= 0x1F)) || ((0x80 <= codePointValue) && (codePointValue <= 0x9F)) || isNaN(codePointValue)) {
-          continue;
-        } else {
-          // String.fromCharCode() is inadequate to handle the entire range of Unicode
-          // Someday after upgrading to ES2015, can use String.fromCodePoint()
-          result += String.kmwFromCharCode(codePointValue);
-        }
-      }
-      return result ? result : null;
-    }
-
-    // Produces a HTMLSpanElement with the key's actual text.
-    protected generateKeyText(vkbd: VisualKeyboard): HTMLSpanElement {
-      let spec = this.spec;
+    public get keyText(): string {
+      const spec = this.spec;
+      const DEFAULT_BLANK = '\xa0';
 
       // Add OSK key labels
-      var keyText;
-      var t = document.createElement('span'), ts=t.style;
+      let keyText = null;
       if(spec['text'] == null || spec['text'] == '') {
-        keyText='\xa0';  // default:  nbsp.
         if(typeof spec['id'] == 'string') {
           // If the ID's Unicode-based, just use that code.
-          keyText = this.unicodeKeyIdToString(spec['id']) || keyText;
+          keyText = keyboards.ActiveKey.unicodeIDToText(spec['id']) || DEFAULT_BLANK;
```

Comment:
```suggestion
          keyText = keyboards.ActiveKey.unicodeIDToText(spec['id']);
```

Line 414 does `|| DEFAULT_BLANK` again anyway.